test/test_torch.py

```diff
@@ -2981,6 +2981,13 @@ def test_arange(self):
         self.assertEqual(r1, r2, 0)
         self.assertEqual(r2, r3[:-1], 0)
 
+        self.assertRaises(RuntimeError, lambda: torch.arange(0, float('nan')))
+        self.assertRaises(RuntimeError, lambda: torch.arange(0, float('-inf')))
+        self.assertRaises(RuntimeError, lambda: torch.arange(0, float('inf')))
```

Comment:
nit: self.assertRaisesRegex (with a regex on unsupported range) would probably be better. I can imagine a (wrong) scenario where we attempt to allocate memory for float('inf') and instead report OOM instead of "unsupported range".

Reply:
fixed